### PR TITLE
Issue #4722: fix SimplifiableIfStatement, MissingPackageInfo  violations

### PR DESCRIPTION
**Pull request by Nimfadora**
Issue #4722

fix SimplifiableIfStatement and MissingPackageInfo  violation

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java [hunk 1]
```diff
@@ -197,14 +197,9 @@ protected boolean hasText(final DetailAST slistAST) {
...
-                    && CommonUtils.isBlank(lastLine)) {
-                // check if all lines are also only whitespace
-                returnValue = !checkIsAllLinesAreWhitespace(lines, slistLineNo, rcurlyLineNo);
-            }
-            else {
-                returnValue = true;
-            }
+            // check if all lines are also only whitespace
+            returnValue = !(CommonUtils.isBlank(firstLine) && CommonUtils.isBlank(lastLine)) ||
+                    !checkIsAllLinesAreWhitespace(lines, slistLineNo, rcurlyLineNo);
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java]: comment is not relevant now.

expression become over complicated to read that is based on negation .... very hard to read.
please revert and use suppression by javadoc tag.

**Comment by Nimfadora** [comment]
@romani if simplification in InnerAssignmentCheck breaks cobertura, so I fix it in the next PR.

**Comment by rnveach** [comment]
> if simplification in InnerAssignmentCheck breaks cobertura

We can't merge a red CI. it will cause issues for everyone until it is fixed.

**Comment by Nimfadora** [comment]
@rnveach you misunderstood me, I didn't introduce simplification, but suppress violation by @noinspection in that case (InnerAssignmentCheck), because if simplification is used - it breaks cobertura. So in this PR everything is ok. In the next I will remove suppression, introduce simplification and add some tests to fix cobertura issue connected with simplification.

File: config/checkstyle_checks.xml [hunk 2]
```diff
@@ -115,7 +115,7 @@
   </module>
   <module name="RegexpOnFilename">
       <property name="folderPattern" value="[\\/]src[\\/]it[\\/]java[\\/]"/>
-      <property name="fileNamePattern" value="^((\w+Test)|(Base\w+))\.java$"/>
+      <property name="fileNamePattern" value="^((\w+Test)|(Base\w+)|(package-info))\.java$"/>
```
Comment (romani) [inline config/checkstyle_checks.xml]: good catch by checkstyle....
We need to suppress MissingPackageInfo in tests, as "package-info" file is required only for javadoc generation, but we do not generate javadoc for tests at all - http://checkstyle.sourceforge.net/apidocs/index.html .
Please use inspection MissingPackageInfo only for "Production" scope in IDEA.
Sorry that I mislead you, system is already smarter than I.

**Comment by Nimfadora** [comment]
@romani MissingPackageInfo generates false-positives, as I can see. [Here](https://github.com/checkstyle/checkstyle/pull/4747/files#diff-589822d701c47b61ccc1ca6a51d86bddR1427) I turned off tests inspection and only enable production code check, but on my local I still have violations on it package and grammars package, and on TC you can see 12 mysterious (it didn't show them) violations. Maybe it should be suppressed.

**Comment by romani** [comment]
I had unstable behavior on my side too, please suppres this inspection completely - "behavior is non stable between TC and local IDEA, disablement in test scope does not work"

**Pull request merged**